### PR TITLE
script

### PR DESCRIPTION
**Pull request by olivakar**
Usage :- 

```
python create_x509_chain.py "leviosa" --ca-password "hogwarts" --intermediate-password "lumos" --device-password "alohomora"
```

### CODE REVIEW

File: azure-iot-device/tools/create_x509_chain.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import os
+import re
+import argparse
```
Comment (cartertinney) [inline azure-iot-device/tools/create_x509_chain.py]: Put in the scripts folder

File: azure-iot-device/tools/create_x509_chain.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        for number, line in enumerate(lines):
+            if re.search(policy_loose_regex, line):
+                policy_loose_found = True
+                break
+
+    list_of_lines = list()
+    ca_default_regex = re.compile(r"\s*\[\s*CA_default\s*\]\s*")
+    ca_default_section_found = False
+    policy_regex = re.compile(r"\s*\s*(policy\s*=\s*policy_strict|policy\s*=\s*policy_match)")
+    with open("C:/OpenSSL-Win64/bin/openssl.cfg", "r") as change_openssl_config:
```
Comment (cartertinney) [inline azure-iot-device/tools/create_x509_chain.py]: >C:/OpenSSL-Win64/bin/openssl.cfg [](start = 15, length = 32)

Only valid on Windows, and with 64 bit at that.

**Comment by cartertinney** [comment]
If you can confirm that it works in both Python 2.7 and 3.5, and ensure that it works on multiple platforms (Windows, OSX, Linux), I'll sign off.

File: azure-iot-device/tools/create_x509_chain.py (hunk 2 above)
Comment (olivakar) [reply azure-iot-device/tools/create_x509_chain.py]: changed to environment variable like the previous one. 

---
In reply to: [297798367](https://github.com/Azure/azure-iot-sdk-python-preview/pull/116#discussion_r297798367) [](ancestors = 297798367)

**Comment by olivakar** [comment]
Confirmed that works on Python 2..
For mac and linux I have to do a little bit more time. 


---
In reply to: [505985870](https://github.com/Azure/azure-iot-sdk-python-preview/pull/116#issuecomment-505985870) [](ancestors = 505985870)

**Comment by pierreca** [comment]
Ha I see why the config file was a problem: you went with invoking the openssl binary directly.
There might be a way to greatly simplify the code (especially all that stuff about dealing with the config file) by using an existing library
have you looked at this for example: [https://cryptography.io/en/latest/x509/](https://cryptography.io/en/latest/x509/)

**Pull request merged**